### ISSUE TITLE
 Clippy release process: Maintaining a beta branch for important uplifts

### ISSUE THREAD

**Issue by Manishearth** [rust-lang/rust-clippy#3957]
Currently we don't uplift false positive fixes to beta, which means that stable can be hit with new lints that still have bugs.

Should we maintain a beta branch? If so, kinds of things qualify for uplifts? We typically have a _lot_ of false positive fixes come in, how do we gauge whether a fix is eligible?

cc @phansch @oli-obk

**Comment by phansch** [comment]
Yeah.. we always have a lot of bug fixes. In the worst case currently, we have to tell people that a fix is going to be included in _~12_ weeks. I agree that it would be nice to include them faster, so a `beta` branch makes sense.

Apart from not riding the whole Rust release train, is there a downside to just including _all_ fixes in this branch?

If it's more work for some reason, we could only do this for fixes that apply to enabled-by-default lints (and ICEs)?

**Comment by Manishearth** [comment]
> Apart from not riding the whole Rust release train, is there a downside to just including all fixes in this branch?

It's annoying to uplift things, some times. Also I'm not sure if we want to cut a new beta for each false positive fix. We can just keep backporting and ask for new betas whenever it's a big bug, maybe.

> If it's too much work, we could only do this for fixes that apply to enabled-by-default lints (and ICEs)?

Sure. But I suspect at least some types of false positive will be pretty niche.
